### ISSUE TITLE
Remove receive hooks for person and person_distinct_id

### ISSUE THREAD

**Issue by macobo** [PostHog/posthog#7489]
## Is your feature request related to a problem?

We currently define post-receive hooks for when person or person_distinct_ids get created or deleted which create/delete the appropriate data in clickhouse.

As far as I'm aware this is widely used in tests for person creation, but some application logic might also rely on it as well.

There are cases where this can cause a lot of trouble:
1. It doesn't handle mass deletions well - if you delete e.g. 100 persons, 100 separate mutations get queued in clickhouse, slowing down queries and potentially blocking migrations or other queries.
2. It's not consistent. Plugin server creates most people and does most of the management around person rows. We shouldn't have a separate out-of-sync mechanism in the app.

## Describe the solution you'd like

Remove the post-receive hooks, update tests and handle clickhouse data deletion asynchronously in a celery task or in a bi-weekly cronjob.

## Describe alternatives you've considered

Never deleting data.

## Additional context

cc @guidoiaquinti @yakkomajuri @fuziontech who were helping to debug an incident today.

**Comment by macobo** [comment]
Another blocker here - demo data generation which does not always go through plugin-server.

**Comment by macobo** [comment]
Other blockers that need to get ported: 
- demo data generation which does not always go through plugin-server.
- posthog/management/commands/generate_revenue_data.py for the same reason
- split_person code path
- Person destroying endpoint
- Team/Organization deletion logic

Re demo data, @EDsCODE you have a WIP branch, will that fix this?

**Comment by EDsCODE** [comment]
demo data won't rely on those hooks anymore but all our tests still rely on the hooks to keep the dbs in sync

**Comment by timgl** [comment]
How would we feel about putting these hooks back for everything except deletes? Right now updating or deleting properties in the interface doesn't work.

I tried doing this without the post-save hooks but I ended up doing stuff like [this](https://github.com/PostHog/posthog/pull/8518/files) which feels silly.

**Comment by Twixes** [comment]
To maintain data integrity (particularly considering migrations) we should _really_ strive to exclusively affect persons (including properties) via events. This leaves us a reproducible almost-transaction log.

**Comment by EDsCODE** [comment]
Adding on to ^: There was a brief point made about this in https://github.com/PostHog/meta/pull/39. Will manually updating person properties even make sense after those changes are made because person properties would be immutable and strictly determined from event interactions?

To make the updating work right now, the hooks could be fine but it seems like we'll need a deprecation plan or different implementation soon anyway

**Comment by timgl** [comment]
> To maintain data integrity (particularly considering migrations) we should really strive to exclusively affect persons (including properties) via events. This leaves us a reproducible almost-transaction log.

Hm yes I guess that makes sense, at least for updating properties. I don't think there's a way of doing this for deleting properties though right?

> Adding on to ^: There was a brief point made about this in https://github.com/PostHog/meta/pull/39. Will manually updating person properties even make sense after those changes are made because person properties would be immutable and strictly determined from event interactions?

This shouldn't be a problem anymore at that point as I assume we'll not be using persons stored in clickhouse anymore

**Comment by Twixes** [comment]
Right, the capture API doesn't define a way of _deleting_ properties – perhaps there should be one. There's `$set: Record<string, any>`, analogously there could be `$unset: string[]`.
